### ISSUE TITLE
v2: findings with bad locations and false positive on missing comments

### ISSUE THREAD

**Issue by noahdietz** [googleapis/api-linter#1530]
> We are getting tons of bugs and false negative warnings with beta 1:
> 
> ```
>     - message: Missing comment over "SetOrganizationIamPolicyResponse". <-- this message does not exists but maybe comes from google.iam.v1.iam_policy.proto somehow
>       location:
>         start_position:
>             line_number: 1
>             column_number: 1
>         end_position:
>             line_number: 1
>             column_number: 0
>         path: mindful/earth/user/v1/organization.proto
>       rule_id: core::0192::has-comments
>       rule_doc_uri: https://linter.aip.dev/192/has-comments
>     - message: Missing comment over "BATCH_GET_USERS_REQUEST_READ_MASK_ACCOUNT".
>       location:
>         start_position:
>             line_number: 1
>             column_number: 1
>         end_position:
>             line_number: 1
>             column_number: 0
>         path: mindful/earth/user/v1/user.proto
>       rule_id: core::0192::has-comments
>       rule_doc_uri: https://linter.aip.dev/192/has-comments
> ```
> 
> for all kind of protos 
> 
> ```
> //mindful/earth/user/membership/v1:membership_v1_proto_api_linter        FAILED in 0.1s
>   /private/var/tmp/_bazel_loeffel/2b8e6c5a281d05be113e22f1346b0da3/execroot/_main/bazel-out/darwin_arm64-fastbuild/testlogs/mindful/earth/user/membership/v1/membership_v1_proto_api_linter/test.log
> //mindful/earth/user/profile/v1:profile_v1_proto_api_linter              FAILED in 0.1s
>   /private/var/tmp/_bazel_loeffel/2b8e6c5a281d05be113e22f1346b0da3/execroot/_main/bazel-out/darwin_arm64-fastbuild/testlogs/mindful/earth/user/profile/v1/profile_v1_proto_api_linter/test.log
> //mindful/earth/user/v1:user_v1_proto_api_linter                         FAILED in 0.1s
>   /private/var/tmp/_bazel_loeffel/2b8e6c5a281d05be113e22f1346b0da3/execroot/_main/bazel-out/darwin_arm64-fastbuild/testlogs/mindful/earth/user/v1/user_v1_proto_api_linter/test.log
> ```
> 
> ```
> sh_test(
>     name = "user_v1_proto_api_linter",
>     srcs = ["@rules//:api_linter"],
>     args = [
>         "mindful/earth/user/v1/user.proto",
>         "mindful/earth/user/v1/organization.proto",
>         "--set-exit-status",
>         "--descriptor-set-in $(location :user_v1_proto_descriptor_set)",
>         "--disable-rule core::0191::java-package",
>         "--disable-rule core::0191::java-multiple-files",
>         "--disable-rule core::0191::java-outer-classname",
>         "--disable-rule core::0157::request-read-mask-field",
>         "--disable-rule core::0134::request-mask-field",
>         "--disable-rule core::0132::request-field-types",
>     ],
>     data = [
>         ":user_v1_proto",
>         ":user_v1_proto_descriptor_set",
>     ],
>     visibility = ["//visibility:private"],
> )
> ```
> 
> no issues with v1.70.0 

 _Originally posted by @loeffel-io in [#1529](https://github.com/googleapis/api-linter/issues/1529#issuecomment-3232578328)_

**Comment by noahdietz** [comment]
root cause for the odd locations (and maybe the "missing comments" findings) is lack of source code info in the descriptors. Bazel proto_library doesn't seem to pass the protoc flag --include_source_info([looks experimental](https://github.com/protocolbuffers/protobuf/blob/4fbd1111a292d04746c732573025e3251de0bb9c/bazel/private/proto_library_rule.bzl#L175-L176) in latest tag?) by default, so passing in a pre-compiled descriptor set to api-linter without SourceCodeInfo means 1. all of the locations will be wrong and 2. comments may be missing. All of our testing is done with that flag enabled as that is what we have in our own usage.

We will look into this and fix it.

**Comment by noahdietz** [comment]
Speaking offline, @quirogas discovered that we have setup `protocompile` resolvers incorrectly to start using the `descriptor-set-in` as a source for the _input_ files, rather than parsing them directly from disk as was done with `protoparse` which had a dedicated `LookupImport` property for the parser.

**Comment by noahdietz** [comment]
@quirogas let us know when your fix is merged and released!

**Comment by quirogas** [comment]
# [v2.0.0-beta.2](https://github.com/googleapis/api-linter/releases/tag/v2.0.0-beta.2)
